perfkitbenchmarker/benchmark_spec.py

```diff
@@ -139,59 +195,35 @@ def ConstructVirtualMachines(self):
     """Constructs the BenchmarkSpec's VirtualMachine objects."""
     vm_group_specs = self.config.vm_groups
 
-    zone_index = 0
+    os_type = vm_group_specs[vm_group_specs.keys()[0]].os_type
+    if os_type == os_types.JUJU:
+        """
+        The Juju VM needs to be created first, so that subsequent units can
+        be properly added under its control.
+        """
+
+        juju_spec = copy.copy(vm_group_specs[vm_group_specs.keys()[0]])
```

Comment:
This probably needs to change as well

Not many benchmarks can make use of it, but different groups can span clouds. Since the controller talks over internal ips, a controller on one cloud wouldn't be able to talk to a group on another cloud. At the very least you need a controller per cloud, but a controller per group might be simpler